### ISSUE TITLE
global variables usable in different js (test) files

### ISSUE THREAD

**Issue by kemalbayar** [cypress-io/cypress#1121]
- Operating System:
- Cypress Version:
- Browser Version:

#### Test code:

is there a way to define variables in order to able to use them 'in different js (test) files'

**Comment by bahmutov** [comment]
Cypress test files are just _regular_ JavaScript files that can `require` and `import` other JS files. For example you can add shared code to your `support` folder `index.js` file https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests.html#Support-file

**Comment by kemalbayar** [comment]
@bahmutov thanks

I tried this before but it didnt work.  I defined alertSeverities variable in commands.js file and when I used that variable in a js file  I encountered with this error ... 
"ReferenceError:  alertSeverities is not defined"

Also, i created a js file for storing my global variables and I indicated its name in index.js file as import './globalvariables' and then I defined alertSeverities variable in globalvariables.js file but when I ran the test it failed again. 

I dont understand why it is not working

**Comment by brian-mann** [comment]
TL;DR - just use modules, not globals.

Cypress is just JavaScript. All of the underlying principles about structuring files apply to Cypress as it would your own application files.

In this case, you keep mentioning **variables**. Variables are things defined in a particular file and are never global. This means it's not possible for them to be shared. Variables are accessible based on the **local scope** in which they are defined.

In order to make them global you have to attach them to a global object: `window`. However, there's no reason to do this, Cypress automatically has module support built in. This enables you to import functions into each spec file, thus making them way more organized and obvious than using globals.

We have recipes of this here: https://docs.cypress.io/examples/examples/recipes.html#Node-Modules

**Comment by kemalbayar** [comment]
Thank you very much for your answers. @brian-mann 

I want to inform you about an issue (if i'm right...) in the following address, 

https://github.com/cypress-io/cypress-example-recipes/blob/master/examples/fundamentals__node-modules/cypress/support/commands.js

Cypress.addParentCommand() {....} used in the command.js file but this is not used anymore, isnt it?

cypress-example-recipes/examples/fundamentals__node-modules/cypress/support/commands.js

Cypress.addParentCommand("login", function(email, password){ ..........

**Comment by chkashif167** [comment]
https://stackoverflow.com/questions/59404379/cypress-variable-scope-outside-the-then-block-i-am-facing-issue
Need help
